nixos/modules/services/misc/linux-enable-ir-emitter.nix

```
@@ -0,0 +1,52 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+with lib; let
+  cfg = config.services.linux-enable-ir-emitter;
+in
+{
+  options = {
+    services.linux-enable-ir-emitter = {
+      enable = mkEnableOption (mdDoc "Linux Enable IR Emitter");
+
+      package = mkPackageOptionMD pkgs "linux-enable-ir-emitter" {} // {
+        description = mdDoc ''
+          Package to use for the Linux Enable IR Emitter service.
+        '';
+      };
+
+      device = mkOption {
+        type = types.lines;
+        default = "video2";
+        description = mdDoc ''
+          Emitter device to depend on.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services.linux-enable-ir-emitter =
+      let
+        targets = [
+          "multi-user.target"
+          "suspend.target"
+          "hybrid-sleep.target"
+          "hibernate.target"
+          "suspend-then-hibernate.target"
+        ];
+      in
+      {
+        description = "Enable the infrared emitter.";
+
+        script = "${getExe cfg.package} run";
+
+        wantedBy = targets;
+        after = targets ++ [ "dev-${cfg.device}.device" ];
+      };
```

```suggestion
    systemd.services.linux-enable-ir-emitter = {
      description = "Enable the infrared emitter.";
      script = "${getExe cfg.package} run";

      wantedBy = [
        "multi-user.target"
        "suspend.target"
        "hybrid-sleep.target"
        "hibernate.target"
        "suspend-then-hibernate.target"
      ];
      after = wantedBy ++ [ "dev-${cfg.device}.device" ];
    };
```
we could make this a bit clearer

---

@SuperSandro2000 are you missing a `rec` on this?